Building under Windows 10 WSL fails

I get syntax errors in build.make files when I build iNav under WSL:
`Built target MATEKF411SE.elf
/bin/sh: 1: Syntax error: "(" unexpected
make[3]: *** [src/main/target/MATEKF411SE/CMakeFiles/MATEKF411SE.dir/build.make:57: src/main/target/MATEKF411SE/CMakeFiles/MATEKF411SE] Error 2
make[2]: *** [CMakeFiles/Makefile2:32144: src/main/target/MATEKF411SE/CMakeFiles/MATEKF411SE.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:32151: src/main/target/MATEKF411SE/CMakeFiles/MATEKF411SE.dir/rule] Error 2
make: *** [Makefile:13131: MATEKF411SE] Error 2`

I found out that the complete Windows environment variable PATH is appended here (is that necessary?), and its not proper escaped:
Example: 
`c/Program\ Files\ (x86)/..` 
Whitespace is escaped, braces not, so make runs into syntax errors.
After adding the escape strings by hand, compilation runs fine.

--- a/docs/development/Building in Windows 10 with Linux Subsystem.md
+++ b/docs/development/Building in Windows 10 with Linux Subsystem.md
@@ -96,7 +96,9 @@ make[1]: *** [CMakeFiles/Makefile2:33290: src/main/target/MATEKF722SE/CMakeFiles
 make: *** [Makefile:13703: MATEKF722SE] Error 2
 ```
 
-This error can be triggered by a Windows PATHs included in the Linux Subsystem. The solution is to:
+This error can be triggered by a Windows PATHs included in the Linux Subsystem. The solution is:
+
+#### For WSL V1 - Flags set as 7 by default
 
 1. Open Windows RegEdit tool
 1. Find `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss\{GUID}\Flags`
@@ -105,3 +107,28 @@ This error can be triggered by a Windows PATHs included in the Linux Subsystem. 
 1. `cd build`
 1. `cmake ..`
 1. `make {TARGET}` should be working again 
+
+#### For WSL V2 - Flags set as 0x0000000f (15) by default
+1. Open Windows RegEdit tool
+1. Find `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss\{GUID}\Flags`
+1. Change `Flags` from `f` to `d`, it is stored as Base Hexadecimal
+1. Restart WSL and Windows preferably
+1. `cd build`
+1. `cmake ..`
+1. `make {TARGET}` should be working again 
+
+#### Or, for either version
+1. In the Linux Subsystem, `cd /etc/`
+2. Create a new file with `sudo nano wsl.conf`
+3. Enter the following in to the new file:
+```
+[Interop]
+appendWindowsPath=false
+```
+4. Save the file by holding `Ctrl` and pressing `o`
+5. Press `Enter` to confirm the wsl.conf filename.
+6. Hit `Ctrl`+`x` to exit nano
+7. Restart WSL and Windows preferably
+8. `cd build`
+9. `cmake ..`
+9. `make {TARGET}` should be working again 

